DfE.Hexagonal.Driver.FunctionApp/Triggers/CreateSchool/CreateSchoolFunction.cs

```diff
@@ -0,0 +1,59 @@
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using DfE.Hexagonal.Application.UseCases.School.CreateSchool;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+
+namespace DfE.Hexagonal.FunctionApp.Triggers.CreateSchool
+{
+    public class CreateSchoolFunction
+    {
+        private readonly ILogger<CreateSchoolFunction> _logger;
+        private readonly ICreateSchool _createSchool;
+        public CreateSchoolFunction(
+            ILogger<CreateSchoolFunction> log,
+            ICreateSchool createSchool)
+        {
+            _logger = log;
+            _createSchool = createSchool;
+        }
+
+        [FunctionName("CreateSchoolFunction")]
+        [OpenApiOperation(operationId: "create-school",
+            tags: new[] { "Create new School" },
+            Summary = "Creates a new school based on the details given",
+            Description = "Need to submit list of UPNs or searchText with optional filters, index type and query type",
+            Visibility = OpenApiVisibilityType.Important)]
+        [OpenApiRequestBody("input", typeof(SchoolInput))]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.Created, contentType: "text/plain", bodyType: typeof(SchoolInput), Description = "The details of the school created")]
+        [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.BadRequest,
+                                   Summary = "Invalid parameters",
+                                   Description = "Either Invalid parameters or parameters are missing")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = Routes.School)] HttpRequest req)
+        {
+            _logger.LogInformation("CreateSchoolFunction processed a request.");
```

Comment:
Log levels - is `information` the correct level to log this at?

Personally I don't have a fully formed view on this yet, but @ScottDawsonDfE might have stronger views/guidance to share?

Reply:
I'd say yes... that is information, may be not something you want logged all the time and I'd want to filter out noise and go straight to Error then work back. Personal preference, happy to hear other views.

Reply:
Information feels like the right level for this, as it is just information about what has happened.
You could argue that it is debug or trace level logging.
If we provided a large amount of logging (such as many logs per method or logging all the minor methods), you would want more differentiation.